### DISCUSSION TITLE
Move Camera Origin in Modelviewer

### DISCUSSION THREAD

**Discussion by LeClam** [google/model-viewer#4733] [Q&A] [answered]
Sorry if this question has been asked already. I have an object with annotations in my modelviewer. 
![Viewport-Illustration](https://github.com/google/model-viewer/assets/134278887/f82974de-e531-4742-a8d9-9f57762d4c49)

When I click on an annotation a sidebar fades in from the right side to show some information.
![Viewport-Illustration2](https://github.com/google/model-viewer/assets/134278887/35d466ea-9990-4a5b-8262-2dc3ec69fd02)
At this point, the modelviewer focusses the Annotation and adjust the camera-view according to the data-orbit and data-target of the annotation button which results in the annotation getting positioned right in the center of the screen. However in case of a sidebar window being visible on the right side, it would cover up the model partially. 

I would therefore highly appreciate it if i could move the "focus" or "origin" slightly to the left
![Viewport-Illustration3](https://github.com/google/model-viewer/assets/134278887/dd635459-40b6-46d6-b7a1-bde597f950b5)

I tried to simply move the whole modelviewer to the side but this of course results in the model getting cropped if I should move it to the far right.
![Viewport-Illustration4](https://github.com/google/model-viewer/assets/134278887/581a51a5-055d-463b-ae4f-d67a092bee44)

Has someone an idea how to solve this?

Thanks in advance

**Comment by elalish** [comment]
I would recommend exactly the approach you took - shrink the MV element. With a slightly different sidebar gradient I think it'll look quite nice. If you're concerned about the model clipping, you can `disable-pan`.

**Reply by LeClam** [reply]
Thanks for the fast response. That's not really the effect I wanted to achieve. The sidebar is semi-transparent and the user should still be able to move the object after clicking on the hotspot. I think you can't make the overflow of the modelviewer "visible" to prevent clipping, right?

Is there no other way to move the origin point in the model viewer?

**Reply by elalish** [reply, marked as answer]
We don't have an API to move the center of the render away from the center of the element. You could probably do some tricks with an over-large element and CSS masking if you really want to.

**Reply by LeClam** [reply]
Oh, that's a bummer, I was hoping something like that would be possible. Anyway thanks for the help.

**Reply by danilokrulj** [reply]
Could this be resolved by setting the target point?

![image](https://github.com/google/model-viewer/assets/92090684/e1065520-d482-427e-9bd6-9a38ee670e7d)
